File: 100-999/111. Minimum Depth of Binary Tree/solution.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
 
class Solution 
{
    public:
        int minDepth(TreeNode* root) 
        {
            if (root)
            {
                int minDepth = 1;

                vector<TreeNode*> nodes = {root};

                while (!nodes.empty())
                {
                    vector<TreeNode*> newNodes;

                    for (auto node : nodes)
                    {
                        if (!node->left && !node->right)
                            return depth;

                        if (node->left)
                            newNodes.push_back(node->left);
                        
                        if (node->right)
                            newNodes.push_back(node->right);
                    }

                    nodes.assign(newNodes.begin(), newNodes.end());
                    minDepth++;
                }

                return minDepth;
            }   

            return 0; 
        }
};

File: 100-999/111. Minimum Depth of Binary Tree/solution.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */

/**
 * @param {TreeNode} root
 * @return {number}
 */

var minDepth = function(root) 
{
    if (root)
    {
        let minDepth = 1
        
        let nodes = [root];

        while (nodes.length)
        {
            const newNodes = [];

            for (const node of nodes) 
            {
                if (!node.left && !node.right)
                    return minDepth

                if (node.left)
                    newNodes.push(node.left)

                if (node.right)
                    newNodes.push(node.right)
            }

            nodes = newNodes
            minDepth++
        }
    }

    return 0
};